### PR TITLE
Add basic layout of Vision Layer

### PR DESCRIPTION
**Pull request by 7ayushgupta**
- Added Gate and Buoy Task
- Added Dynamic Reconfiguration files
- Added the OpenCV basic library

### CODE REVIEW

**Comment by Mayankm96** [comment]
It is normally advised that you break your PR to smaller PRs each having a fewer files changes since reviewing them would be easy. Few general styling issues:
1. the metadata has to be cleaned of all comments. Keep only the main comments required over there (see previous packages in the repo to understand). 
2. README needs to be added for each package that is being sent to the PR so that it is easy to understand them in future
3. I suppose you are using 8 spaces tab in your IDE. It would be better to switch it to 4 instead since rest of the code uses that
4. The bracketing is an issue. Follow the style-guide [here](https://github.com/AUV-IITK/AUVWiki/wiki/Developers-Guide#bracketing)
5. The source code is lacking comments a lot! You would need to add them otherwise it ould be difficult in future to understand the code. Try to keep the openCV and the ros part of the code separate as we had discussed that day.

Make these changes first. After that I'll check for compilation and other issues (if any).

File: control_layer/level_mode_controller/include/level_mode_controller/Thruster.h.orig [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#ifndef Thruster_H
```
Comment (Mayankm96) [inline control_layer/level_mode_controller/include/level_mode_controller/Thruster.h.orig]: This field is redundant. Modify the `.gitignore` files to ignore such files.

**Comment by Mayankm96** [comment]
There are several `.orig` files that have been created. Also, do we even need level mode controller/ Why not remove it? I think Rithvik opened an issue on the same. Also, I am closing this PR. Please send in your pull requests with a small number of files at a time. Typically package wise. There are 24 commits already which is too much in my opinion. 

@ksakash Comments?

**Comment by 7ayushgupta** [comment]
Yes, I am going to close this one. No point creating such a big PR. We'll send in two smaller PRs for both the packages. We'll also fix the redundant code issue.

**Pull request closed without merging**